apps/server/src/modules/files-storage/uc/files-storage-restore.uc.spec.ts

```diff
@@ -0,0 +1,293 @@
+import { AuthorizationService } from '@src/modules/authorization';
+import { S3ClientAdapter } from '../client/s3-client.adapter';
+import { FileRecordParams, SingleFileParams } from '../controller/dto';
+import { FileRecordParentType } from '../entity';
+import { PermissionContexts } from '../files-storage.const';
+import { FilesStorageMapper } from '../mapper';
+import { FilesStorageService } from '../service/files-storage.service';
+import { FilesStorageUC } from './files-storage.uc';
+
+const getFileRecordsWithParams = () => {
+	const userId = new ObjectId().toHexString();
+	const schoolId = new ObjectId().toHexString();
+
+	const fileRecords = [
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text.txt' }),
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-two.txt' }),
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-tree.txt' }),
+	];
+
+	const params: FileRecordParams = {
+		schoolId,
+		parentId: userId,
+		parentType: FileRecordParentType.User,
+	};
+
+	return { params, fileRecords, userId };
+};
+
+const getFileRecordWithParams = () => {
+	const userId = new ObjectId().toHexString();
+	const schoolId = new ObjectId().toHexString();
+
+	const fileRecord = fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text.txt' });
+
+	const params: SingleFileParams = {
+		fileRecordId: fileRecord.id,
+	};
+
+	return { params, fileRecord, userId };
+};
+
+describe('FilesStorageUC', () => {
+	let module: TestingModule;
+	let filesStorageUC: FilesStorageUC;
+	let filesStorageService: DeepMocked<FilesStorageService>;
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let orm: MikroORM;
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		module = await Test.createTestingModule({
+			providers: [
+				FilesStorageUC,
+				{
+					provide: S3ClientAdapter,
+					useValue: createMock<S3ClientAdapter>(),
+				},
+				{
+					provide: FilesStorageService,
+					useValue: createMock<FilesStorageService>(),
+				},
+				{
+					provide: AntivirusService,
+					useValue: createMock<AntivirusService>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: HttpService,
+					useValue: createMock<HttpService>(),
+				},
+			],
+		}).compile();
+
+		filesStorageUC = module.get(FilesStorageUC);
+		authorizationService = module.get(AuthorizationService);
+		filesStorageService = module.get(FilesStorageService);
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+	});
+
+	afterAll(async () => {
+		await orm.close();
+	});
+
+	afterEach(async () => {
+		await module.close();
+	});
+
+	it('should be defined', () => {
+		expect(filesStorageUC).toBeDefined();
+	});
+
+	describe('restoreFilesOfParent is called', () => {
+		describe('WHEN user is authorised', () => {
```

Comment:
..and files for restoring exists  ..but the mocks pass only files that are not marked for delete. Maybe it make sense to add a helper to factory to create directly already marked for deleting files ...see examples in task factory.

Same in line 187.